lonestar/CMakeLists.txt

```diff
@@ -23,6 +23,36 @@ function(add_test_scale type app)
   endforeach()
 endfunction(add_test_scale)
 
+function(app_cpu PREFIX name bin_name)
+  string(CONCAT target_name ${PREFIX} ${bin_name})
```

Comment:
FWIW, the original idea behind getting rid of the CMake `app` function was to reduce the chance of introducing inadvertent coupling through a shared `app` function.

Also, it turns out most of the build configuration for a lonestar app is exactly the same as building any application in CMake so hiding this behind a function might be more confusing for people who just want to use Galois libraries in their own projects.

Anyway, I can see the value of reducing duplication in the CMake files but I wanted to raise the fact that I went down this path before and wanted to go the other way because `app` was making things more brittle.

Reply:
Sure. I also agreed with you, but didn't know how to balance the reducing-codes and the inadvertent code couplings. Also as you can see, some scientific apps and the gpu apps required separate codes from the functions, which looked very messy.

For this issue, I would like to hear opinions from distributed and GPU Galois teams since they maintained the style coupling the codes through the functions.
If all agree with that,  I could decouple the codes from app_cpu(), app_dist(), and app_gpu(). 
Please correct if I misunderstood.

Reply:
This is debatable. It depends on the app or set of apps. The question is whether we want to be consistent. In any case, it should be a separate issue/commit/pull-request.

Reply:
I agree. Let's not introduce these extra functions. The duplication is small and it makes the CMake configuration files much harder to read for someone unfamiliar with the project.

Reply:
So,
1) Do you guys want to handle this issue on the separate pull or issue?
If 1) is yes, Then 2) Do you guys want to keep the shared function for now? or want to separate them?

Please confirm those two questions. Then I could restart this request.

Reply:
If you can, keep it separate. If you have already done something, you don't need to undo it. I would suggest merging this as soon as possible. So try to minimize the changes you do.

Reply:
OK